include/CL/opencl.hpp

```diff
@@ -10253,6 +10342,530 @@ namespace compatibility {
     };
 } // namespace compatibility
 
+#if defined(cl_khr_command_buffer)
+/*! \class CommandBufferKhr
+ * \brief CommandBufferKhr interface for cl_command_buffer_khr.
+ */
+class CommandBufferKhr : public detail::Wrapper<cl_command_buffer_khr>
+{
+private:
+    static std::once_flag default_initialized_;
+    static CommandBufferKhr default_;
+    static cl_int default_error_;
```

Comment:
I think that the use of std::once_flag suggests that the code is thread-safe , because these static variables happens only once..

Reply:
This is not what I was thinking to but more what is the goal of this static variables at the first place? What happened if 2 command buffers are used at the same time in different threads
Please add a Doxygen-like comment on each of them explaining what they are for, otherwise the reader might think they are useless. ;-)

Reply:
I actually don't think it makes much sense to have a "default" command buffer like it does to have a default context or command queue, and I'd suggest removing all of the "default" command buffer handling.

Reply:
OK